### PR TITLE
Handle the union of never types in function return

### LINKED ISSUE #39797: [Bug]: Empty member types in `BUnionType` for return type `never|never`
### Description

When defining the return type of a method as `never|never` the compiler resolves it as a `BUnionType` with its  'memberTypes' linked hash set being empty. However in `originalMemeberTypes` one `BNeverType` member is present.  

prob

### Steps to Reproduce

```
public function main() {
    f3();
}

function f3() returns never|never {}
```

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### LINKED ISSUE #39834: [Bug]: Inconsistent errors in function invocation and definition with functions that return `never`
### Description

From the below examples, for `() a = d();` we get the following errors:

> unreachable code. 
> expression of type 'never' or equivalent to type 'never' not allowed here.

But for `() b = f();` we get the following errors:

> unreachable code.
> this function must return a result.

Similarly for `function d() returns never {}` we get:

> function with return type 'never' or equivalent to type 'never' cannot implicitly return 'nil' by falling off the end of the function body

But for `function f() returns never|never {}` we get the following error

> this function must return a result

These scenarios should ideally give similar error messages.

### Steps to Reproduce

```
public function main() {
    () a = d();
    () b = f();
}

function d() returns never {} 
function f() returns never|never {}
```

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

Could be related to #39797

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose

Fixes #39834
Fixes #39797

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java [hunk 1]
```diff
@@ -200,6 +202,10 @@ public static BUnionType create(BTypeSymbol tsymbol, LinkedHashSet<BType> types)
             }
         }
 
+        if (!foundNotNeverType) {
+            memberTypes.addAll(types);
+        }
```
[the remark this answers is not in the document]
Comment (chiranSachintha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java:207]:
```ballerina
type Foo Bar|never;
type Bar never|never;
```
If we define the `Foo` type as above, we include both `Bar` and `never` types as a member types of `Foo`.

File: tests/jballerina-unit-test/src/test/resources/test-src/types/never/never_code_analysis_negative.bal [hunk 2]
```diff
@@ -67,3 +67,31 @@ function testCyclicInclusionViaFieldWithNever() {
     Foo _ = {}; // error: expression of type 'never' or equivalent to type 'never' not allowed here
     Bar _ = {x: {}}; // OK
 }
+
+function testReturnUnionOfNever1() returns never|never {
```
[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply tests/jballerina-unit-test/src/test/resources/test-src/types/never/never_code_analysis_negative.bal:71]: Added

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java]: > If the type is `string | never`, we will create a union type by including only the `string` type as a member. If the type is `never | never`, we will add the never type as a member. Why do we treat these two scenarios differently?

As per the spec, we can consider `string|never` as `string`, but it is incorrect to consider  `never|never` as no type.

[the remark this answers is not in the document]
Comment (KavinduZoysa) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java]: > ```ballerina
> type Foo Bar|never;
> type Bar never|never;
> ```
> 
> If we define the `Foo` type as above, we include both `Bar` and `never` types as a member types of `Foo`.

Fixed

**Review by gimantha — APPROVED** [review]

**Pull request merged**